src/schemas.ts

```diff
@@ -160,6 +160,522 @@ const foxglove_Pose: FoxgloveMessageSchema = {
+      description: "Drawing primitive to use for triangles",
+    },
+    {
+      name: "pose",
+      type: { type: "nested", schema: foxglove_Pose },
+      description: "Origin of triangles relative to reference frame",
+    },
+    {
+      name: "points",
+      type: { type: "nested", schema: foxglove_Point3 },
+      array: true,
+      description: "Vertices to use for triangles",
+    },
+    {
+      name: "color",
+      type: { type: "nested", schema: foxglove_Color },
+      description:
+        "Solid color to use for the whole shape. One of `color` or `colors` must be provided.",
+    },
+    {
+      name: "colors",
+      type: { type: "nested", schema: foxglove_Color },
+      array: true,
+      description:
+        "Per-vertex colors (if specified, must have the same length as `points`). One of `color` or `colors` should be provided.",
+    },
+    {
+      name: "indices",
+      type: { type: "primitive", name: "uint32" },
+      array: true,
+      description:
+        "Indexes into the `points` and `colors` attribute arrays, which can be used to avoid duplicating attribute data.",
+    },
+  ],
+};
+
+const foxglove_ModelMarker: FoxgloveMessageSchema = {
+  type: "message",
+  name: "ModelMarker",
+  description: "A marker representing a 3D model",
+  fields: [
+    ...commonMarkerFields,
+    {
+      name: "pose",
+      type: { type: "nested", schema: foxglove_Pose },
+      description: "Origin of model relative to reference frame",
+    },
+    {
+      name: "scale",
+      type: { type: "nested", schema: foxglove_Vector3 },
+      description: "Scale factor to apply to the model along each axis",
+    },
+    {
+      name: "color",
+      type: { type: "nested", schema: foxglove_Color },
+      description:
+        "Solid color to use for the whole model. If `use_embedded_materials` is true, this color is blended on top of the embedded material color.",
+    },
+    {
+      name: "use_embedded_materials",
+      type: { type: "primitive", name: "boolean" },
+      description: "Whether to use materials embedded in the model, or only the `color`",
+    },
+    {
+      name: "url",
+      type: { type: "primitive", name: "string" },
+      description:
+        "URL pointing to model file. Either `url` or `mime_type` and `data` should be provided.",
+    },
+    {
+      name: "mime_type",
+      type: { type: "primitive", name: "string" },
+      description:
+        "MIME type of embedded model (e.g. `model/gltf-binary`). Either `url` or `mime_type` and `data` should be provided.",
+    },
+    {
+      name: "data",
+      type: { type: "primitive", name: "bytes" },
+      description: "Embedded model. Either `url` or `mime_type` and `data` should be provided.",
+    },
+  ],
+};
+
+const foxglove_Markers: FoxgloveMessageSchema = {
+  type: "message",
+  name: "Markers",
+  description: "A list that can contain any number of any type of marker",
+  fields: [
```

Comment:
Why don't we remove `ImageAnnotations` and make this new `Markers` list support all markers?

Reply:
> Why don't we remove `ImageAnnotations` and make this new `Markers` list support all markers? I don't think it really matters whether the topic contains 2D or 3D markers or both?

I think it does. The ImageAnnotations unit values are in pixels and they are designed around annotating an image. They don't typically stand alone without an image under them. "Markers" are designed around scene building and the units are in scene space (world coordinates). Markers make sense on their own since they are the building blocks of a scene even without sensor data under them.